### PR TITLE
Retry for failed pushes

### PR DESCRIPTION
**Pull request by sgugger**
This PR changes the doc building jobs to try to rebase and push again if the first push fails.

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Sure, let's give this a try!

File: .github/workflows/build_pr_documentation.yml [hunk 1]
```diff
@@ -89,7 +89,9 @@ jobs:
           if [[ `git status --porcelain` ]]; then
             git add .
             git commit -m "Updated with commit ${{ inputs.commit_sha }} See: https://github.com/huggingface/${{ inputs.package }}/commit/${{ inputs.commit_sha }}"
-            git push origin main
+            git push origin main ||
+            (echo "Failed on the first try, rebasing and pushing again" && git pull --rebase git && push origin main) ||
+            (echo "Failed on the second try, rebasing and pushing again" && git pull --rebase git && push origin main)
```
Comment (julien-c) [inline .github/workflows/build_pr_documentation.yml:94]: `&&` not at the right place?

Comment (julien-c) [reply .github/workflows/build_pr_documentation.yml:94]: yes they call me eagle eye

File: .github/workflows/build_main_documentation.yml [hunk 2]
```diff
@@ -134,7 +136,9 @@ jobs:
             if [[ `git status --porcelain` ]]; then
               git add ${{ inputs.notebook_folder }}
               git commit -m "Updated ${{ inputs.package }} doc notebooks with commit ${{ inputs.commit_sha }} \n\nSee: https://github.com/huggingface/${{ inputs.package }}/commit/${{ inputs.commit_sha }}"
-              git push origin master
+              git push origin main ||
+              (echo "Failed on the first try, rebasing and pushing again" && git pull --rebase && git push origin master) ||
+              (echo "Failed on the second try, rebasing and pushing again" && git pull --rebase && git push origin master)
```
Comment (julien-c) [inline .github/workflows/build_main_documentation.yml:141]: master or main?

Comment (sgugger) [reply .github/workflows/build_main_documentation.yml:141]: Can we rename all those masters to main already ;-)

File: .github/workflows/build_pr_documentation.yml (hunk 1 above)
Comment (sgugger) [reply .github/workflows/build_pr_documentation.yml:94]: Oh good catch, thanks a lot!

File: .github/workflows/build_main_documentation.yml (hunk 2 above)
Comment (LysandreJik) [reply .github/workflows/build_main_documentation.yml:141]: Let's rename `transformers`' `master` branch to `main` on Monday? :smiley:

Comment (sgugger) [reply .github/workflows/build_main_documentation.yml:141]: Let's goooooooooooooooooooooooooooo!

Comment (mishig25) [reply .github/workflows/build_main_documentation.yml:141]: Necessary changes for the doc moon-landing:
https://github.com/huggingface/moon-landing/pull/2409
https://github.com/huggingface/doc-build/pull/6

**Pull request merged**